internal/dag/accessors.go

```diff
@@ -108,116 +109,193 @@ func externalName(svc *v1.Service) string {
 	return svc.Spec.ExternalName
 }
 
-// GetSecureVirtualHosts returns all secure virtual hosts in the DAG.
-func (dag *DAG) GetSecureVirtualHosts() map[ListenerName]*SecureVirtualHost {
-	getter := svhostGetter(map[ListenerName]*SecureVirtualHost{})
-	dag.Visit(getter.visit)
-	return getter
-}
-
 // GetSecureVirtualHost returns the secure virtual host in the DAG that
 // matches the provided name, or nil if no matching secure virtual host
 // is found.
-func (dag *DAG) GetSecureVirtualHost(ln ListenerName) *SecureVirtualHost {
-	return dag.GetSecureVirtualHosts()[ln]
+func (d *DAG) GetSecureVirtualHost(hostname string) *SecureVirtualHost {
+	return d.SecureVirtualHosts[hostname]
 }
 
 // EnsureSecureVirtualHost adds a secure virtual host with the provided
 // name to the DAG if it does not already exist, and returns it.
-func (dag *DAG) EnsureSecureVirtualHost(ln ListenerName) *SecureVirtualHost {
-	if svh := dag.GetSecureVirtualHost(ln); svh != nil {
+func (d *DAG) EnsureSecureVirtualHost(hostname string) *SecureVirtualHost {
+	if svh := d.GetSecureVirtualHost(hostname); svh != nil {
 		return svh
 	}
 
 	svh := &SecureVirtualHost{
 		VirtualHost: VirtualHost{
-			Name:         ln.Name,
-			ListenerName: ln.ListenerName,
+			Name:         hostname,
+			ListenerName: "ingress_https",
```

Comment:
nit: maybe we can use a constant here? there is one in the `internal/xdscache/v3` package but not sure that makes sense to use that

Reply:
yeah absolutely. I was gonna do a follow-up PR to move the ListenerName onto the Listener struct, so I'll add/use a const as part of that. Need to double-check on the many listeners thing though to make sure I'm not undoing past work here.